modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -0,0 +1,211 @@
+    {
+        return false;
+    }
+
+    virtual bool allowsMerge(const cv::gimpl::GIslandModel::Graph &,
+                             const ade::NodeHandle &,
+                             const ade::NodeHandle &,
+                             const ade::NodeHandle &) const override
+    {
+        return false;
+    }
+};
+
+GStreamingExecutable::GStreamingExecutable(const ade::Graph& g,
+                                           const std::vector<ade::NodeHandle>& nodes)
+    : m_g(g), m_gm(m_g)
+{
+    using namespace cv::gimpl;
+    const auto is_op = [this](const ade::NodeHandle &nh)
+    {
+        return m_gm.metadata(nh).get<NodeType>().t == NodeType::OP;
+    };
+
+    auto it = std::find_if(nodes.begin(), nodes.end(), is_op);
+    GAPI_Assert(it != nodes.end() && "No operators found for this island?!");
+
+    ConstStreamingGraph cag(m_g);
+    m_actor = cag.metadata(*it).get<StreamingCreateFunction>().createActorFunction();
+
+    // Ensure this the only op in the graph
+    if (std::any_of(it+1, nodes.end(), is_op))
+    {
+        cv::util::throw_error
+            (std::logic_error
+             ("Internal error: Streaming subgraph has multiple operations"));
+    }
+}
+
+
+} // anonymous namespace
+
+cv::gapi::GBackend cv::gapi::streaming::backend()
+{
+    static cv::gapi::GBackend this_backend(std::make_shared<GStreamingBackendImpl>());
+    return this_backend;
+}
+
+cv::gapi::GKernelPackage cv::gapi::streaming::kernels()
+{
+    return cv::gapi::kernels<cv::gimpl::Copy, cv::gimpl::BGR>();
+}
+
+void cv::gimpl::Copy::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
+                                 cv::gimpl::GIslandExecutable::IOutput &out)
+{
+    while (true)
+    {
+        const auto in_msg = in.get();
+        if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg))
+        {
+            out.post(cv::gimpl::EndOfStream{});
+            return;
+        }
+
+        const cv::GRunArgs &in_args = cv::util::get<cv::GRunArgs>(in_msg);
+        GAPI_Assert(in_args.size() == 1u);
+
+        cv::GRunArgP out_arg = out.get(0);
+        *cv::util::get<cv::MediaFrame*>(out_arg) = cv::util::get<cv::MediaFrame>(in_args[0]);
+        out.post(std::move(out_arg));
+    }
+}
+
+void cv::gimpl::BGR::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
+                                cv::gimpl::GIslandExecutable::IOutput &out)
+{
+    while (true)
+    {
+        const auto in_msg = in.get();
+        if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg))
+        {
+            out.post(cv::gimpl::EndOfStream{});
+            return;
+        }
+
+        const cv::GRunArgs &in_args = cv::util::get<cv::GRunArgs>(in_msg);
+        GAPI_Assert(in_args.size() == 1u);
+
+        cv::GRunArgP out_arg = out.get(0);
+        auto frame = cv::util::get<cv::MediaFrame>(in_args[0]);
+        const auto& desc = frame.desc();
+
+        auto& rmat = *cv::util::get<cv::RMat*>(out_arg);
```

Comment:
Likely your output `RMat` is allocated already according to our graph metadata.

Probably we should avoid this by _not_ allocating output `RMat` for this backend - since allocating that buffer is completely unnecessary and nobody will actually use it, while for 4K it may be a costly allocation.

cc: @rgarnov

Reply:
@rgarnov any comments here?

Reply:
As far as I know, I override `allocateOutputs` method, so `RMat` shouldn't be allocated

Reply:
There weren't `allocate` functions, added them. Checked locally that they are called